src/main/java/hachi/Parser.java

```diff
@@ -0,0 +1,219 @@
+     * @param argumentLength The number of arguments which the user passed in.
+     * @throws HachiException The appropriate exception if argument length is wrong.
+     */
+    public static void checkArgumentLength(String commandWord, int argumentLength)
+            throws HachiException {
+        switch (commandWord) {
+        case ByeCommand.COMMAND_WORD:
+            if (argumentLength > 0) {
+                throw new TooManyArgumentsException(ByeCommand.COMMAND_WORD, 0, argumentLength);
+            }
+            break;
+        case ListCommand.COMMAND_WORD:
+            if (argumentLength > 0) {
+                throw new TooManyArgumentsException(ListCommand.COMMAND_WORD, 0, argumentLength);
+            }
+            break;
+        case MarkCommand.COMMAND_WORD:
+            if (argumentLength > 1) {
+                throw new TooManyArgumentsException(MarkCommand.COMMAND_WORD, 1, argumentLength);
+            }
+            if (argumentLength < 1) {
+                throw new EmptyNumberException(MarkCommand.COMMAND_WORD);
+            }
+            break;
+        case UnmarkCommand.COMMAND_WORD:
+            if (argumentLength > 1) {
+                throw new TooManyArgumentsException(UnmarkCommand.COMMAND_WORD, 1, argumentLength);
+            }
+            if (argumentLength < 1) {
+                throw new EmptyNumberException(UnmarkCommand.COMMAND_WORD);
+            }
+            break;
+        case DeleteCommand.COMMAND_WORD:
+            if (argumentLength > 1) {
+                throw new TooManyArgumentsException(DeleteCommand.COMMAND_WORD, 1, argumentLength);
+            }
+            if (argumentLength < 1) {
+                throw new EmptyNumberException(DeleteCommand.COMMAND_WORD);
+            }
+            break;
+        case TodoCommand.COMMAND_WORD:
+            if (argumentLength < 1) {
+                throw new EmptyTaskException(TodoCommand.COMMAND_WORD);
+            }
+            break;
+        case DeadlineCommand.COMMAND_WORD:
+            if (argumentLength < 1) {
+                throw new EmptyTaskException(DeadlineCommand.COMMAND_WORD);
+            }
+            break;
+        case EventCommand.COMMAND_WORD:
+            if (argumentLength < 1) {
+                throw new EmptyTaskException(EventCommand.COMMAND_WORD);
+            }
+            break;
+        case SearchdateCommand.COMMAND_WORD:
+            if (argumentLength > 1) {
+                throw new TooManyArgumentsException(SearchdateCommand.COMMAND_WORD, 1, argumentLength);
+            }
+            if (argumentLength < 1) {
+                throw new DateFormatWrongException("");
+            }
+            break;
+        case FindCommand.COMMAND_WORD:
+            if (argumentLength < 1) {
+                throw new HachiException("Please include the word or words you are searching for!");
+            }
+            break;
+        default:
+            break;
```

Comment:
I was thinking the default case should do something like printing out errors rather than just a break statement.